nntrainer/layers/cl_layers/reshape_cl.cpp

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Niket Agarwal <niket.a@samsung.com>
+ *
+ * @file   reshape_cl.cpp
+ * @date   18 June 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Niket Agarwal <niket.a@samsung.com>
+ * @bug	   No known bugs except for NYI items
+ * @brief  This is Reshape GPU Layer Implementation
+ */
+
+#include <iostream>
+#include <layer_context.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <reshape_cl.h>
+
+std::string reshape_cl_kernel_fp16_ =
+  R"(
+    #pragma OPENCL EXTENSION cl_khr_fp16 : enable
+    __kernel void reshape_cl_fp16(__global const half* input, 
+                               __global half* output,
+                               const int batchsize, 
+                               const int channels, 
+                               const int height, 
+                               const int width) {
+
+    int elements_per_batch = channels * height * width;
+    int global_id = get_global_id(0);
+    int batch_index = global_id / elements_per_batch;
+    int element_index = global_id % elements_per_batch;
+    
+    if (batch_index < batchsize) {
+        int input_channel = element_index / (height * width);
+        int input_height = (element_index % (height * width)) / width;
+        int input_width = element_index % width;
+        
+        int input_index = batch_index * channels * height * width +
+                          input_channel * height * width +
+                          input_height * width +
+                          input_width;
+
+        int output_index = batch_index * elements_per_batch + element_index;
+        
+        output[output_index] = input[input_index];
+    }
+})";
+
+std::string reshape_cl_kernel_ =
+  R"(__kernel void reshape_cl(__global const float* input, 
+                               __global float* output,
+                               const int batchsize, 
+                               const int channels, 
+                               const int height, 
+                               const int width) {
+
+    int elements_per_batch = channels * height * width;
+    int global_id = get_global_id(0);
+    int batch_index = global_id / elements_per_batch;
+    int element_index = global_id % elements_per_batch;
+    
+    if (batch_index < batchsize) {
+        int input_channel = element_index / (height * width);
+        int input_height = (element_index % (height * width)) / width;
+        int input_width = element_index % width;
+        
+        int input_index = batch_index * channels * height * width +
+                          input_channel * height * width +
+                          input_height * width +
+                          input_width;
+
+        int output_index = batch_index * elements_per_batch + element_index;
+        
+        output[output_index] = input[input_index];
+    }
+})";
```

Comment:
if the whole process is simply copying, couldn't it be as follows?

```
int i= get_global_id(0);
output[i] = input[i]
```

Reply:
Thanks for the suggestion! I have tested and updated in the latest commit.